JASP-Desktop/widgets/listmodelfiltereddataentry.cpp

```diff
@@ -370,23 +377,26 @@ void ListModelFilteredDataEntry::setColName(QString colName)
 	if (_colName == colName)
 		return;
 
-	//std::cout << "ListModelFilteredDataEntry::setColName(" << colName.toStdString() << ") editableColumn = "  << _editableColumn /*<< " en _colNames[edit] = " << (_colNames.size() > _editableColumn ? _colNames[_editableColumn].toStdString() : "Not there" ) */ << std::endl;
-
-	bool changed = _colName != colName;
+	if (_colName.isEmpty())
+	{
+		_colNames.push_back(colName);
+		_editableColumn = _colNames.size() - 1;
+	}
+	else if (colName.isEmpty())
+	{
+		if (_colNames.size() > 0) _colNames.pop_back();
+		_editableColumn = -1;
+	}
+	else if (_colNames.size() > _editableColumn && _editableColumn >= 0)
```

Comment:
What happens if `_editableColumn == -1` ?
Is this even a possibility?

If it isn't this check is unnecessary and if it is, well then there needs to be an else right?

Reply:
So looking at the file I don't think _editableColumn can be <0 if a colName was set before and will be set now right?

Reply:
Yes, if _colName is empty, then _editableColumn is -1. I find always a good idea to always check the index before using it in a array: in this way, even if there is a bug, it does not crash at least.
But if we want to remove the redundancy of information, I could beter remove the _editableColumn property: if _colName is empty it is always -1, and if not, it is always the last index of the _colNames array. This would remove the possibility to have an editable column which is not the last one... but ja, this TableView has to be refactored anyway.

Reply:
See https://github.com/jasp-stats/INTERNAL-jasp/issues/1098

Reply:
But that means that if `_colName == ""` at the start of the function then `_editableColumn` is -1.
So if `colName != ""` then it gets set to `_colNames.size() - 1`.
If `colName == ""` then we don't even get to this if-else-if-else-if. Because then `_colName == colName` and we would've `return`ed already.

My point is, including `_editableColumn >= 0` is pointless, and when I read that code I wonder "Why are we checking for this? Is it possible?" But it isn't.

So then I would prefer either a crash, or at the very least and  `else Log::log() << "Shit is crazy WTF? _editableColumn == -1! How is it possible?" << std::endl;`

And refactoring would be glorious